pydantic_ai_slim/pydantic_ai/output.py

```diff
@@ -0,0 +1,289 @@
+    print(repr(result.output))
+    #> Vehicle(name='Ford Explorer', wheels=4)
+    ```
+    """
+
+    outputs: Sequence[OutputTypeOrFunction[OutputDataT]]
+    name: str | None
+    description: str | None
+    template: str | None
+    """Template for the prompt passed to the model.
+    The '{schema}' placeholder will be replaced with the output JSON schema.
+    If not specified, the default template specified on the model's profile will be used.
+    """
+
+    def __init__(
+        self,
+        type_: OutputTypeOrFunction[OutputDataT] | Sequence[OutputTypeOrFunction[OutputDataT]],
+        *,
+        name: str | None = None,
+        description: str | None = None,
+        template: str | None = None,
+    ):
+        self.outputs = _flatten_output_spec(type_)
+        self.name = name
+        self.description = description
+        self.template = template
+
+
+@dataclass
+class TextOutput(Generic[OutputDataT]):
+    """Marker class to use text output for an output function taking a string argument.
+
+    Example:
+    ```python
+    from pydantic_ai import Agent, TextOutput
+
+
+    def split_into_words(text: str) -> list[str]:
+        return text.split()
+
+
+    agent = Agent(
+        'openai:gpt-4o',
+        output_type=TextOutput(split_into_words),
+    )
+    result = agent.run_sync('Who was Albert Einstein?')
+    print(result.output)
+    #> ['Albert', 'Einstein', 'was', 'a', 'German-born', 'theoretical', 'physicist.']
+    ```
+    """
+
+    output_function: TextOutputFunction[OutputDataT]
+
+
+def _get_union_args(tp: Any) -> tuple[Any, ...]:
+    """Extract the arguments of a Union type if `output_type` is a union, otherwise return an empty tuple."""
+    if typing_objects.is_typealiastype(tp):
+        tp = tp.__value__
+
+    origin = get_origin(tp)
+    if is_union_origin(origin):
+        return get_args(tp)
+    else:
+        return ()
+
+
+def _flatten_output_spec(output_spec: T | Sequence[T]) -> list[T]:
+    outputs: Sequence[T]
+    if isinstance(output_spec, Sequence):
+        outputs = output_spec
+    else:
+        outputs = (output_spec,)
+
+    outputs_flat: list[T] = []
+    for output in outputs:
+        if union_types := _get_union_args(output):
+            outputs_flat.extend(union_types)
+        else:
+            outputs_flat.append(output)
+    return outputs_flat
+
+
+T_co = TypeVar('T_co', covariant=True)
+
+OutputTypeOrFunction = TypeAliasType(
+    'OutputTypeOrFunction', Union[type[T_co], Callable[..., Union[Awaitable[T_co], T_co]]], type_params=(T_co,)
+)
+
+OutputSpec = TypeAliasType(
+    'OutputSpec',
+    Union[
+        OutputTypeOrFunction[T_co],
+        ToolOutput[T_co],
+        ModelStructuredOutput[T_co],
+        PromptedStructuredOutput[T_co],
+        TextOutput[T_co],
+        Sequence[Union[OutputTypeOrFunction[T_co], ToolOutput[T_co], TextOutput[T_co]]],
+    ],
+    type_params=(T_co,),
+)
+
+TextOutputFunction = TypeAliasType(
```

Comment:
again, I'm not clear which of these are really public and and which aren't.

Reply:
They're all types of public fields.